app/controllers/users_controller.rb

```diff
@@ -0,0 +1,36 @@
+class UsersController < ApplicationController
+  def new
+    @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    # @user = User.new(username: params[:username], email: params[:email], password: params[:password])
+    if @user.save
+      redirect_to new_user_path
+    else
+      render :new
+    end
```

Comment:
# __OPTIONAL__
### instead of using `if-else` you could have placed it in 2 lines only using the following code:
```ruby
return redirect_to new_user_path if @user.valid
render :edit
```
I hope it makes sense and will serve you for futures notes :wink: